code/game/objects/structures/extinguisher.dm

```diff
@@ -142,13 +146,56 @@
 			new /obj/item/wallframe/extinguisher_cabinet(loc)
 		else
 			new /obj/item/stack/sheet/metal (loc, 2)
-		if(stored_extinguisher)
-			stored_extinguisher.forceMove(loc)
-			stored_extinguisher = null
+		if(stored_object)
+			stored_object.forceMove(loc)
+			stored_object = null
 	qdel(src)
 
 /obj/item/wallframe/extinguisher_cabinet
 	name = "extinguisher cabinet frame"
 	desc = "Used for building wall-mounted extinguisher cabinets."
 	icon_state = "extinguisher"
 	result_path = /obj/structure/extinguisher_cabinet
+
+//wall mounted medkits
+/obj/structure/extinguisher_cabinet/medkit
+	name = "medkit cabinet"
+	desc = "A small wall mounted cabinet designed to hold a first aid kit."
+	icon = 'icons/obj/wallmounts.dmi'
+	icon_state = "medkit_closed"
+	var/obj/item/storage/firstaid/regular/stored_medkit
+	//sets allowed item for the cabinet. Setting it to main type will allow subtypes.
+	alloweditem = /obj/item/storage/firstaid
+
+//mapcode
+/obj/structure/extinguisher_cabinet/medkit/Initialize(mapload, ndir, building)
+	. = ..()
+	if(building)
+		setDir(ndir)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -27 : 27)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0

```

Comment:
Make the pixel_x/pixel_y thing variables on the item and get rid of this entire duplicate code block.

Reply:
No longer applies. Anyway that just sets which side of the wall its on if I understand.